src/traffic_server/CoreUtils.cc

```diff
@@ -495,11 +467,10 @@ CoreUtils::load_http_hdr(HTTPHdr *core_hdr, HTTPHdr *live_hdr)
   HdrHeap *heap_ptr                 = (HdrHeap *)http_hdr->m_heap;
   intptr_t ptr_heaps                = 0;
   intptr_t ptr_heap_size            = 0;
-  intptr_t ptr_xl_size              = 2;
   intptr_t str_size                 = 0;
   intptr_t str_heaps                = 0;
   MarshalXlate default_MarshalXlate = {nullptr, nullptr, nullptr};
-  DynArray<struct MarshalXlate> ptr_xlation(&default_MarshalXlate, 2);
+  std::vector<struct MarshalXlate> ptr_xlation(2, default_MarshalXlate);
```

Comment:
This isn't going to suffice. The declaration of `MarshalXlate` needs to be changed to initialize the pointers to `nullptr`, otherwise the later `resize` will potentially do bad things. It's not going to use a default object like this, but the default constructor.

Reply:
I think I can change the resize to something like `ptr_xlation.resize(..., default_MarshalXlate)` so resize won't break;

Reply:
I think fixing the default constructor be correct is better than hacking around it.

Reply:
Sure